src/main/java/org/alfresco/trashcan/TrashcanCleaner.java

```diff
@@ -93,110 +90,71 @@ public TrashcanCleaner(NodeService nodeService, TransactionService transactionSe
     {
-        List<ChildAssociationRef> childAssocs = getTrashcanChildAssocs();
-        List<NodeRef> nodes = new ArrayList<>(deleteBatchCount);
-        if (logger.isDebugEnabled())
-        {
-            logger.debug(String.format("Found %s nodes on trashcan", childAssocs.size()));
-        }
-        return fillBatchToDelete(nodes, childAssocs);
-    }
-
-    /**
-     * 
-     * It will fill up a {@link java.util.List List} of
-     * {@link org.alfresco.service.cmr.repository.NodeRef NodeRef} from all the
-     * {@link org.alfresco.service.cmr.repository.ChildAssociationRef
-     * ChildAssociationRef} of the archive store set, according to the limit
-     * parameters: <b>deleteBatchCount</b> and <b>daysToKeep</b>.
-     * 
-     * @param batch
-     * @param trashChildAssocs
-     * @return
-     */
-    private List<NodeRef> fillBatchToDelete(List<NodeRef> batch, List<ChildAssociationRef> trashChildAssocs)
-    {
-        for (int j = trashChildAssocs.size(); j > 0 && batch.size() < deleteBatchCount; j--) 
-        {
-            ChildAssociationRef childAssoc = trashChildAssocs.get(j - 1);
-            NodeRef childRef = childAssoc.getChildRef();
-            if (olderThanDaysToKeep(childRef))
-            {
-                batch.add(childRef);
-            }
-        }
-        return batch;
+        return getTrashcanChildAssocs()
+                .stream()
+                .filter(el -> olderThanDaysToKeep(el.getChildRef()))
+                .map(ChildAssociationRef::getChildRef)
+                .collect(Collectors.toList());
     }
 
     /**
-     * 
-     * It will return all
-     * {@link org.alfresco.service.cmr.repository.ChildAssociationRef
-     * ChildAssociationRef} of the archive store set.
-     * 
+     *
+     * It will return the first {@link #deleteBatchCount}
+     * {@link org.alfresco.service.cmr.repository.ChildAssociationRef}s
+     * of type {@link ContentModel}.ASSOC_CHILDREN
+     * from the archive store set.
+     *
      * @return
      */
     private List<ChildAssociationRef> getTrashcanChildAssocs()
     {
         StoreRef archiveStore = new StoreRef(archiveStoreUrl);
         NodeRef archiveRoot = nodeService.getRootNode(archiveStore);
-        List<ChildAssociationRef> allChildren = nodeService.getChildAssocs(archiveRoot);
-        return filterArchiveUsers(allChildren);
-    }
-
-    /**
-     * 
-     * Don't include on list of nodes to be deleted the archiveuser node types.
-     * 
-     * @return
-     */
-    private List<ChildAssociationRef> filterArchiveUsers(List<ChildAssociationRef> allChilds)
-    {
-        List<ChildAssociationRef> children = new ArrayList<>();
-        for (ChildAssociationRef childAssoc : allChilds) 
-        {
-            NodeRef child = childAssoc.getChildRef();
-            if (!ContentModel.TYPE_ARCHIVE_USER.equals(nodeService.getType(child)))
-            {
-                children.add(childAssoc);
-            }
```

Comment:
Why did we remove this filtering?

Reply:
We filter the nodes by association type [here](https://github.com/Alfresco/alfresco-trashcan-cleaner-module/pull/7/files#diff-481b1ed71300f831eb4955aa52c43d3a31d7a51dcd322695c23709523fa285e4R154). This way we get only the oldest _deleteBatchCount_ nodes in trashcan of type ContentModel.ASSOC_CHILDREN.